tests/raftstore/pd.rs

```diff
@@ -505,39 +664,50 @@ impl TestPdClient {
         let region = self.get_region_by_id(region_id).wait().unwrap();
         panic!("region {:?} has peer {:?}", region, peer);
     }
+
     pub fn add_region(&self, region: &metapb::Region) {
         self.cluster.wl().add_region(region)
     }
 
+    pub fn ensure_transfer_leader(&self, region_id: u64, peer: metapb::Peer) {
+        let op = Operator::TransferLeader {
+            peer,
+            policy: SchedulePolicy::TillSuccess,
+        };
+        self.schedule_operator(region_id, op);
+    }
+
     pub fn add_peer(&self, region_id: u64, peer: metapb::Peer) {
-        self.set_rule(box move |region: &metapb::Region, _: &metapb::Peer| {
-            debug!(
-                "[region {}] trying add {:?} to {:?}",
-                region_id, peer, region
-            );
-            if region.get_id() != region_id {
-                return None;
-            }
-            new_pd_add_change_peer(region, peer.clone())
-        });
+        let op = Operator::AddPeer {
+            peer: Either::Left(peer),
+            policy: SchedulePolicy::Repeat(1),
```

Comment:
Why use 1?

Reply:
In previous implementation, `AddPeer` is not only scheduled once.

Reply:
Yes. In previous,  it's scheduled when TiKV heartbeats PD periodically. Now we schedule it on every polling. So they are essentially the same.

Reply:
But the policy here is `Repeat(1)`, how can they be the same?

Reply:
So, it seems you should replace `add_peer` and `remove_peer` with `SchedulePolicy::TillSuccess`, and leave `must_add_peer` and `must_remove_peer` unchanged.

Reply:
Some tests expect add-peer failure due to unsafe conf change, so I think repeat 1 is enough, `TillSuccess` will repeat the operator till it is success.